GVFS/GVFS.Common/Maintenance/PackfileMaintenanceStep.cs

```diff
@@ -127,6 +127,12 @@ protected override void PerformMaintenance()
                         activity.RelatedWarning($"Skipping {nameof(PackfileMaintenanceStep)} due to git pids {string.Join(",", processIds)}", Keywords.Telemetry);
                         return;
                     }
+
+                    if (this.Context.Repository.HasActiveLibGit2Repo)

```

Comment:
Could this be a race condition?  I.e. it's false here and then turns true before GetPackFilesInfo

Reply:
We can't stop libgit2 from becoming active during our git process, so we could still get IDX files. We'll try to delete them, and the next time we come around we will clean them up (because libgit2 has disposed in the meantime and dropped the handles).

Reply:
That makes sense, but it's a bit confusing to have a check that isn't bulletproof.

Maybe add a comment explaining or just remove the check?  As you said we'll recover on the next run.  There are ways to make the check more bulletproof but probably overkill for our case.

Reply:
The only way to guarantee we won't hit libgit2 issues is to block use of the `LibGit2RepoInvoker` while this step is running. That is unacceptable as the invoker is used to hydrate the file system.

This is similar to how we don't run if there are active `git` processes. It keeps us from running twice while the same libgit2 repo is open across both runs. That allows us to clean up any stale IDX files created in the previous run.

Reply:
Ah I see it's another fine as long as it didn't run across two runs check.  

Perhaps add a short comment saying something to effect of:
"If LibGit2Repo is active it may be spanning two calls to PackfileMaintenceStep which would be problamatic.  This check verifies that condition hasn't occurred."

That would remove any confusion and make it clear the condition we're trying to guard against.  I really should have added a similar comment in LooseObjectsStep.cs.  I'll make sure I'll do that next time I'm in that code.